### PR TITLE
chore(api): dockerize api

### LINKED ISSUE #33: [API] Dockerize API
Also, mention steps to run it in the README.

### PR DESCRIPTION
**Pull request by Aksh-Bansal-dev**
closes #33

### CODE REVIEW

**Review by Samy-33 — CHANGES_REQUESTED** [review]
Later on, we want to include `nginx` into the api image. It will provide users with the environment which will mimic the deployed version better.

File: api/dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# base image  
+FROM python:3.8   
+
+# setup environment variable  
+ENV DockerHOME=/home/app/webapp
```
Comment (Samy-33) [inline api/dockerfile:5]: I'd make environment variables CAPITALIZED with underscores separating words
So `DockerHOME` becomes: `DOCKER_HOME`

**Review by Samy-33 — CHANGES_REQUESTED** [review]

File: api/dockerfile [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+# where your code lives  
+WORKDIR $DockerHOME  
+
+# set environment variables  
+ENV PYTHONDONTWRITEBYTECODE 1
+ENV PYTHONUNBUFFERED 1  
+
+# install dependencies  
+RUN pip install --upgrade pip && pip install -r requirements.txt
```
Comment (Samy-33) [inline api/dockerfile:18]: This should be run after we copy all the files to $DOCKER_HOME.

**Review by Samy-33 — CHANGES_REQUESTED** [review]

File: api/dockerfile [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+# copy whole project to your docker home directory. 
+COPY . $DockerHOME  
+# run this command to install all dependencies  
+RUN pip install -r requirements.txt  
+RUN python manage.py makemigrations
+# port where the Django app runs  
+EXPOSE 8000  
+# start server  
+CMD python manage.py runserver 0.0.0.0:8000
```
[the remark this answers is not in the document]
Comment (Aksh-Bansal-dev) [reply api/dockerfile:30]: Should I do something like this?
```
ENTRYPOINT /usr/local/bin/python manage.py runserver 0.0.0.0:8000
```

[the remark this answers is not in the document]
Comment (Samy-33) [reply api/dockerfile:30]: From the [docs](https://docs.docker.com/engine/reference/builder/#entrypoint), it should be something like:
```Dockerfile
ENTRYPOINT ['python', 'manage.py', 'runserver']
```

**Review by Samy-33 — DISMISSED** [review]
LGTM.

nit: Some comments in the dockerfile do not match with the command they were written for.
For example:
- `# set working directory` is supposed to explain `RUN mkdir -p $API_HOME`, but the command just makes a directory and doesn't set the working directory.
- `# where you code lives` is not clear enough. Which code are you talking about and which place? I'd go with something like: `# Working directory where the api code is stored inside the container`

nit^2: Should we rename the `dockerfile` to `Dockerfile` which I believe is the convention?

**Review by Samy-33 — APPROVED** [review]
LGTM

**Pull request merged**